Create ApplyHistoryView

mun9769 opened the pull request:
## 🔥 *Pull requests*

⛳️ **작업한 브랜치**
- #124 

👷 **작업한 내용**
 - [x] ApplyHistoryView 만들기
 - [x] LTCICard 만들기 (HomeView 가운데 있는 카드)
 - [x] 로고를 터치하면 초기화하기 (ui 초기화, pdf 초기화)
 - [x] noti 아이콘 누르면 LTCICard가 flip 돼요

## 추가 작업
- [x] 전송 완료 후 HomeView State 값 변경하기 (커밋 예정)
- [x] 초기화 버튼이 model도 초기화시켜주기 (커밋 예정)
## 🚨 참고 사항
- layout만 잡은 상태입니다. 디자인은 추후 넣을게요

## 📸 스크린샷
|LTCICard|ApplyHistoryView|
|:--:|:--:|
|<img src = "https://github.com/DeveloperAcademy-POSTECH/MacC-Team2-Nutty/assets/59304977/184fe067-d11d-43be-887a-8c7d395dddba" width=180>|<img src = "https://github.com/DeveloperAcademy-POSTECH/MacC-Team2-Nutty/assets/59304977/147cdf0b-319f-4903-b187-42e597791c78" width=180>|

mun9769 committed:
[Feat/#124] create LTCICard

HomeView의 state가 확장될 수 있기 때문에 bool 변수로 관리하지 않고
enum을 만들어주었다

HomeViewState enum이름 추천좀.. 지금은 ready, apply라고 지음

mun9769 committed:
[Feat/#124] reset mainview

온봄 로고를 누르면 초기화된다

mun9769 committed:
[Feat/#124] create ApplyHistoryView

테스트를 위해 patient, agent mock 데이터를 만들었다
스타일은 아직.,,

mun9769 committed:
[Fix/#124] unimport kakao package

JJunghyunY commented on Onbom/Onbom/Model/Address.swift:
```diff
@@ -9,4 +9,17 @@
+    
+    init() {}
+    
+    init(_ cityAddress: String, _ detailAddress: String) {
```
P3. 이니셜라이저에 기본값을 넣어주면 더 좋지 않을까요?

mun9769 replied:
변수 선언에서 이미 초기화 값이 있어서 안넣어줬어요

JJunghyunY commented on Onbom/Onbom/Model/Address.swift:
```diff
@@ -9,4 +9,17 @@
+        self.detailAddress = detailAddress
+    }
+    
+    func toString() -> String {
```
P3. computed property로 구현하면 어떨까요

mun9769 replied:
```swift
    var toString: String {
        get { return "\(cityAddress)\(detailAddress)" }
    }
```
문법이 swifty 하네요 맘에 듭니다

JJunghyunY replied:
exactly

JJunghyunY commented on Onbom/Onbom/View/LTCI/HomeView.swift:
```diff
@@ -158,6 +145,124 @@ struct HomeView: View {
+        }
+    }
+    
+    private func onReset() {
```
P3. Model값 초기화도 해주어야할까요?

mun9769 replied:
Patient와 Agent에 reset 함수를 추가해야 버그를 방지할 수 있을거 같네요. 곧 커밋올릴게요

JJunghyunY replied:
exactly!!

JJunghyunY reviewed (approved):
ApplyHistoryView 제작하시느라 고생하셨네요
온보딩을 HomeNavigation에 포함시킨건, 버튼클릭으로 앱내에서 온보딩이 보이게끔 초기화하기 위함이겠죠

JJunghyunY commented:
번외로 어제 말씀하신 symbol이 아닌 image asset의 tint color를 바꾸는걸 찾아보니 renderingMode를 이용하면 되더군요
좋은거 알아갑니다 👍

mun9769 committed:
[Chore/#126] modify computed property

more swifty, more powerful

mun9769 committed:
[Feat/#124] reset model

mun9769 committed:
[Feat/#124] change the homeviewState after you apply LTCI

homeviewmodel을 만들고, SubmitView가 homeviewmodel를 의존한다.
단, SubmitView는 SubmitCheckListView를 통해 생성되기 때문에
SubmitListView도 의존성을 가져야 된다는 문제가 있다

mun9769 committed:
Merge branch 'dev' into 124-create-applyhistoryview

mun9769 committed:
[Design/#124] design LTCICard, ApplyHistoryView

mun9769 committed:
Merge branch '124-create-applyhistoryview' of https://github.com/DeveloperAcademy-POSTECH/MacC-Team2-Nutty into 124-create-applyhistoryview
